### ISSUE TITLE
Incorrect line number shown in some error messages

### ISSUE THREAD

**Issue by ZornsLemma** [stardot/beebasm#38]
```
$ cat -n bad-line-number.beebasm
     1	org &2000
     2	.start
     3	
     4	macro foo
     5		for i, 0, 2
     6			nop
     7		next
     8	
     9		lda #10
    10		ldx (&70),y
    11	endmacro
    12	
    13	foo
    14	
    15	.end
    16	
    17	save "foo", start, end
$ ./beebasm -i bad-line-number.beebasm 
bad-line-number.beebasm:9: error: Indirect mode not allowed for this instruction.

ldx (&70),y
    ^

Call stack:
bad-line-number.beebasm:13
```

The "Indirect mode not allowed for this instruction." error shows line 9, not line 10.

**Comment by ZornsLemma** [comment]
This seems to be due to blank lines in macros being completely ignored. If you delete the blank line 8 from the example above, the problem doesn't happen. If you add more blank lines, the line number in the error message becomes increasingly incorrect.

I have a fix for this on https://github.com/ZornsLemma/beebasm/tree/incorrect-line-number, which also adds a couple of test cases in the examples directory. I will test it a bit over the next few weeks and see if it causes any other problems It would be great if someone felt like reviewing the change. I'm not totally happy with it, it feels like there should be a more elegant solution but I really didn't want to mess with the complex logic in LineParser::ProcessLine() more than necessary.

**Comment by ZornsLemma** [comment]
I've merged this to proposed-updates.

**Comment by mungre** [comment]
Fixed in v1.10
